src/pkg/dockerhub/dockerhub.go

```diff
@@ -0,0 +1,219 @@
+package dockerhub
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/docker/cli/cli/config"
+)
+
+var ErrNoCredentials = errors.New("no Docker Hub credentials found in docker config")
+var ErrInvalidCredential = errors.New("invalid Docker Hub credentials")
+
+func GenerateNewPAT(ctx context.Context, label string) (string, string, error) {
+	// Same function used by docker cli config.LoadDefaultConfigFile(cli.err)
+	// https://github.com/docker/cli/blob/master/cli/config/config.go#L166
+	// Which was called in cli initialization here:
+	// https://github.com/docker/cli/blob/511dad69d0733d0b63e5ef637e67b5bf3f7081f0/cli/command/cli.go#L251
+	var errbuf bytes.Buffer
+	cfg := config.LoadDefaultConfigFile(&errbuf)
+	if errbuf.Len() > 0 {
+		return "", "", fmt.Errorf("failed to load docker config: %s", errbuf.String())
+	}
+
+	// Lookup auth entry for the dockerhub registry
+	// The registry host index.docker.io is from https://github.com/moby/moby/blob/v28.5.2/registry/config.go#L49
+	// To avoid issue with unable to import local package by moby/moby
+	auth, err := cfg.GetAuthConfig("https://index.docker.io/v1/")
+	if err != nil {
+		return "", "", fmt.Errorf("failed to get auth config for Docker Hub: %v", err)
+	}
+
+	if auth.Username == "" || auth.Password == "" {
+		return "", "", ErrNoCredentials
+	}
+
+	// TODO: Check PAT scopes to decide if we need to create a new one with minimal scopes
+	var pat string
+	if !strings.HasPrefix(auth.Password, "dckr_pat_") {
+		// Create a new DockerHub PAT
+		var docHubClient DockerHubClient
+		if err := docHubClient.Login(ctx, auth.Username, auth.Password); err != nil {
```

Comment:
What is the user experience here? Does the user get prompted for anything? Does the browser open?